### PR TITLE
resolved spaces in directory name in path

### PR DESCRIPTION
**Pull request by moghya**
as GitHub doesn't allow spaces in directory name GitHub API was raising an exception and hence those spaces are replaced by '-' character.
Regular expression is used and lambda function to replace in all directory names is implemented

### CODE REVIEW

**Comment by Enteleform** [comment]
@moghya 

I just ran into the same issue you mentioned in #559, and have found that the issue is caused by other characters as well.

Specifically:
```python
[
  (' ', '%20'),
  ('`', '%60'),
  ('@', '%40'),
  ('#', '%23'),
  ('$', '%24'),
  ('%', '%25'),
  ('^', '%5E'),
  ('&', '%26'),
  ('=', '%3D'),
  ('+', '%2B'),
  ('[', '%5B'),
  (']', '%5D'),
  ('{', '%7B'),
  ('}', '%7D'),
  (';', '%3B'),
  (',', '%2C'),
]
```

&nbsp;  
`urllib.parse` seems like it should handle all of the above cases, which would the resolve the encoding issues:
```python
import urllib

def encode_path(path:str):
  return urllib.parse.quote(path)

def decode_path(path:str):
  return urllib.parse.unquote(path)
```

File: github/Repository.py [hunk 1]
```diff
@@ -1303,7 +1304,9 @@ def create_file(self, path, message, content,
             put_parameters["author"] = author._identity
         if committer is not github.GithubObject.NotSet:
             put_parameters["committer"] = committer._identity
-
+        
+        #for replacing one or more spaces in directory names specified in  path with  '-' 
+        path =  '/'.join([re.sub(' +','-',directory) for directory in path.split('/')[0:-1]]+[path.split('/')[-1]])
```
Comment (htgoebel) [inline github/Repository.py]: - This is hard to understand. What exactly does this code do?
- It looks like you are replacing spaces in all path parts but the last one. But why not in the last part?
- Why are you replace *multiple* spaces with a *single* dash?
- Also you are splitting the whole string twice.

**Comment by htgoebel** [comment]
ACk, this looks like the URL should be encoded properly using urlencode, if `requests.requestJsonAndCheck()` does not do this.

File: github/Repository.py (hunk 1 above)
Comment (moghya) [reply github/Repository.py]: Replacing spaces with dash as github does .
Github doesn't allow spaces in directory names. It automatically converts spaces into dashes.

Not converting spaces in last part as last part would be filename and spaces in filename are allowed.


Yea, splitting string twice should not he done.
My bad there.

**Comment by sfdye** [comment]
What's the behavior of uploading a file to github via web if the filename contains space? Without some unit tests, it's very hard for me to merge this.

**Pull request closed without merging**